src/main/java/org/opensearch/flowframework/util/ParseUtils.java

```diff
@@ -413,4 +424,21 @@ public static Map<String, String> parseJsonFileToStringToStringMap(String path)
         Map<String, String> mappedJsonFile = mapper.readValue(jsonContent, Map.class);
         return mappedJsonFile;
     }
+
+    /**
+     * Takes an input string, then checks if there is an array in the string with backslashes around strings
+     * (e.g.  "[\"text\", \"hello\"]" to "["text", "hello"]"), this is needed for processors that take in string arrays,
+     * This also removes the quotations around the array making the array valid to consume
+     * (e.g. "weights": "[0.7, 0.3]" to "weights": [0.7, 0.3])
+     * @param input The inputString given to be transformed
+     * @return the transformed string
+     */
+    public static String removingBackslashesAndQuotesInArrayInJsonString(String input) {
+        return Pattern.compile("\"\\[(.*?)]\"").matcher(input).replaceAll(matchResult -> {
```

Comment:
using regex to manipulate JSON strings can be risky due to the complexity and nested nature of JSON. Regex might not handle all cases correctly, especially with nested arrays or objects. Consider whether a JSON parsing library could more reliably achieve the goal.

Reply:
we probably want a more narrow regex to define that these are an array of numerics

Reply:
This compilation might fail for the cases where there might be multiple occurrences of quoted JSON arrays in the input string.
Can you try the below?
```java
Pattern pattern = Pattern.compile("\"\\[(.*?)]\"");
Matcher matcher = pattern.matcher(input);

 StringBuffer result = new StringBuffer();
 while (matcher.find()) {
               // Extract matched content and remove backslashes before quotes
            String withoutEscapes = matcher.group(1).replaceAll("\\\\\"", "\"");
            // Return the transformed string with the brackets but without the outer quotes
            matcher.appendReplacement(result, "[" + withoutEscapes + "]");
}
         // Append remaining input after the last match
    matcher.appendTail(result);
    
    return result.toString();

```

Or else use a JSON parsing library as Jackie called out.

Reply:
yes this does have limitations, for example this fails on the parsing:
```
{
    "create_ingest_pipeline.model_id": "UyE4XY4BDyI7YxgR_A1v",
    "normalization-processor.combination.parameters.weights": "[0.3, 0.7, #$2_)]"
}
```

for example tho if we try to create a hybrid search processor with nested array it actaully gets cast exception on the processor code itself:
```
{
    "create_ingest_pipeline.model_id": "UyE4XY4BDyI7YxgR_A1v",
    "normalization-processor.combination.parameters.weights": "[0.3, 0.7, [0.3]]"
}
```
fails with:
```
java.lang.ClassCastException: class java.util.ArrayList cannot be cast to class java.lang.Double (java.util.ArrayList and java.lang.Double are in module java.base of loader 'bootstrap')
        at java.base/java.util.stream.ReferencePipeline$3$1.accept(ReferencePipeline.java:197) ~[?:?]
        at java.base/java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1625) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:509) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:499) ~[?:?]
        at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateSequential(ReduceOps.java:921) ~[?:?]
        at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234) ~[?:?]
        at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:682) ~[?:?]
        at org.opensearch.neuralsearch.processor.combination.ScoreCombinationUtil.getWeights(ScoreCombinationUtil.java:42) ~[?:?]
        ```

Reply:
I wasn't sure if other library methods would have other unexpected results since I basically tested it with the two main use cases of having a string array and a numerical array and while in the file its all in quotes, when we give to the create pipeline transport actions we don't want those. Do you suggest finding some backslash remover in jackson or somewhere else?

>we probably want a more narrow regex to define that these are an array of numerics

there's also the case where its not just numeric like here:     "create_search_pipeline.retrieval_augmented_generation.context_field_list": "[\"text\"]"

Reply:
Additionally this is only being used in createPipeline steps where we are just limiting to what we can expect, I can also try and find a library that will solve all cases more generically

Reply:
changed to suggestion Owais gave